examples/material_gallery/lib/demo/buttons_demo.dart

```diff
@@ -0,0 +1,210 @@
+      new _ButtonDemo(title: 'DROPDOWN', text: _dropdownText, builder: buildDropdownButton)
+    ];
+  }
+
+  Widget buildFloatingButton(BuildContext context) {
+    return new SizedBox(
+      height: 128.0,
+      child: new Center(
+        child: new FloatingActionButton(
+          child: new Icon(icon: 'content/add'),
+          onPressed: () {
+            Navigator.push(context, new MaterialPageRoute(builder: (_) => new TabsFabDemo()));
+          }
+        )
+      )
+    );
+  }
+
+  Widget buildRaisedButton(BuildContext context) {
+    return new Container(
+      margin: const EdgeDims.symmetric(vertical: 16.0),
+      child: new Column(
+        children: <Widget>[
+          new RaisedButton(
+            child: new Text("Launch Demo"),
+            onPressed: () {
+              Navigator.push(context, new MaterialPageRoute(builder: (_) => new SnackBarDemo()));
+            }
+          ),
+          new RaisedButton(
+            child: new Text("Disabled")
+          )
+        ]
+        .map((Widget child) {
+          return new Container(
+            margin: const EdgeDims.symmetric(vertical: 8.0),
+            child: child
+          );
+        })
+        .toList()
+      )
+    );
+  }
+
+  Widget buildFlatButton(BuildContext context) {
+    return new Container(
+      margin: const EdgeDims.symmetric(vertical: 16.0),
+      child: new ButtonTheme(
+        color: ButtonColor.accent,
+        child: new Column(
+          children: <Widget>[
+            new FlatButton(
+              child: new Text("Launch Demo"),
+              onPressed: () {
+                showDialog(context: context, child: new DialogDemo()).then((String value) {
+                  if (value != null) {
+                    Scaffold.of(context).showSnackBar(new SnackBar(
+                      content: new Text('You dismissed the dialog with "$value"')
+                    ));
+                  }
+                });
+              }
+            ),
+            new FlatButton(
+              child: new Text("Disabled")
+            )
+          ]
+          .map((Widget child) {
+            return new Container(
+              margin: const EdgeDims.symmetric(vertical: 8.0),
+              child: child
+            );
+          })
+          .toList()
+        )
+      )
+    );
+  }
+
+  String dropdownValue = "Free";
+
+  Widget buildDropdownButton(BuildContext context) {
+    return new SizedBox(
+      height: 256.0,
+      child: new Center(
+        child: new DropDownButton<String>(
+          value: dropdownValue,
+          onChanged: (String newValue) {
+            setState(() {
+              if (newValue != null)
+                dropdownValue = newValue;
+            });
+          },
+          items: <String>["One", "Two", "Free", "Four"].map((String value) {
+            return new DropDownMenuItem<String>(
+              value: value,
+              child: new Text(value));
+          })
+          .toList()
```

Comment:
This style, which looks particularly good when there's a cascade of .foo()s from from the Dart style guide: https://www.dartlang.org/effective-dart/style/#do-place-the--on-the-next-line-in-a-multi-line-expression

I realize that we don't follow the Dart Style in general. I've thought that this particular bit helped readability.